libs/sdk-core/src/models.rs

```diff
@@ -132,63 +134,144 @@ pub struct ReverseSwapPairInfo {
 }
 
 /// Details of past or ongoing reverse swaps, as stored in the Breez local DB
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ReverseSwapInfo {
-    // static immutable data
-    pub claim_address: String,
+    /// The reverse swap ID, as reported by the Boltz API in case of a successful creation
+    pub id: String,
+
+    /// Local timestamp when the reverse swap is created
+    pub created_at: i64,
+
+    /// Locally generated preimage, revealed in the last step of the reverse swap
+    pub local_preimage: Vec<u8>,
+
+    /// Locally generated private key, used to sign the claim tx
+    pub local_private_key: Vec<u8>,
+
+    /// On-chain destination address, to which the reverse swap will finally send funds to
+    pub destination_address: String,
+
+    /// Reverse swap status, as reported by the Boltz API. Automatically updated on every new block.
+    pub boltz_api_status: BoltzApiReverseSwapStatus,
+
+    // Only available in the first response from Boltz API:
+    pub hodl_bolt11: String,
+    pub redeem_script: String,
+
+    pub cache: ReverseSwapInfoCached,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ReverseSwapInfoCached {
     pub lockup_address: String,
+    pub onchain_amount_sat: u64,
+}
 
-    // dynamic data
-    pub status: ReverseSwapStatus,
+impl ReverseSwapInfo {
+    /// Dynamically determine the status, based on the cached Boltz API status.
+    ///
+    /// In case of [BoltzApiReverseSwapStatus::SwapCreated], the payment status is looked up on the LN node,
+    /// to distinguish between a reverse swap that is still ongoing and one where the HODL payment failed.
+    pub async fn status(&self, persister: Arc<SqliteStorage>) -> Result<ReverseSwapStatus> {
```

Comment:
It seems that we don't need the ReverseSwapStatus as a struct that maps from boltz status as it is only used internally. How about using directly the boltz status and get rid of this struct?

Reply:
We still need other information (in addition to the Boltz status) to determine the status of a rev swap.

For example:

- Boltz status `swap.created` + paying the HODL invoice failed = we should track it as failed
- Boltz status `swap.created` + paying the HODL invoice succeded = we should track it as in progress

I added a simplified `ReverseStatusSwap` in 0c344187ca0e7f004e2d2428caac30585522a969 :

https://github.com/breez/breez-sdk/blob/0c344187ca0e7f004e2d2428caac30585522a969/libs/sdk-core/src/models.rs#L309-L326